### PR TITLE
Rewrite login dialog

### PR DESCRIPTION
**Pull request by TheOneRing**
OAuth | BasicAuth
--- | ---
image | image
image | image

### CODE REVIEW

File: src/gui/creds/httpcredentialsgui.cpp [hunk 1]
```diff
@@ -109,93 +104,72 @@ void HttpCredentialsGui::asyncAuthResult(OAuth::Result r, const QString &token,
...
-
-    dialog->setTopLabelText(tr("Please enter your password to log in to the account %1.").arg(_account->displayNameWithHost()));
-
-    auto *contentWidget = qobject_cast<BasicLoginWidget *>(dialog->contentWidget());
-    contentWidget->forceUsername(user());
+    auto basicCredentials = new QmlBasicCredentials(_account->hostName(), _account->davDisplayName(), this);
+    basicCredentials->setReadOnlyName(user());
+    _modalWidget = new AccountModalWidget(tr("Login required"), QUrl(QStringLiteral("qrc:/qt/qml/org/ownCloud/gui/qml/credentials/BasicAuthCredentials.qml")),
+        basicCredentials, ocApp()->gui()->settingsDialog());
+    basicCredentials->setParent(_modalWidget);
```
Comment (TheOneRing) [inline src/gui/creds/httpcredentialsgui.cpp:115]: the ownersheip is a bit complicated...

File: src/gui/creds/qmlcredentials.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+{
+    return _userName;
+}
+
+void QmlBasicCredentials::setUserName(const QString &userName)
+{
+    if (_userName != userName) {
+        _userName = userName;
+        Q_EMIT userNameChanged();
+        Q_EMIT readyChanged();
```
Comment (erikjv) [inline src/gui/creds/qmlcredentials.cpp:86]: Why do you emit this signal? The ready boolean didn't changed, nor did the `_oauth` property. Maybe this is intended behaviour, but please document it.

File: src/gui/creds/qmlcredentials.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+{
+    return _password;
+}
+
+void QmlBasicCredentials::setPassword(const QString &password)
+{
+    if (_password != password) {
+        _password = password;
+        Q_EMIT passwordChanged();
+        Q_EMIT readyChanged();
```
Comment (erikjv) [inline src/gui/creds/qmlcredentials.cpp:100]: Same as with `setUserName`

File: src/gui/creds/qmlcredentials.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+bool OCC::QmlBasicCredentials::isReadOnlyName() const
+{
+    return _readOnlyName;
+}
+
+bool QmlBasicCredentials::ready() const
+{
+    return !_userName.isEmpty() && !_password.isEmpty();
```
Comment (erikjv) [inline src/gui/creds/qmlcredentials.cpp:117]: I'm confused, why is there a `_ready` property?

File: src/libsync/httplogger.cpp [hunk 5]
```diff
@@ -154,7 +154,7 @@ void HttpLogger::logRequest(QNetworkReply *reply, QNetworkAccessManager::Operati
             if (ctx->lastUrl != reply->url()) {
                 ctx->addRedirect(reply->url());
             } else {
-                Q_UNREACHABLE();
+                // Q_UNREACHABLE();
```
Comment (erikjv) [inline src/libsync/httplogger.cpp:157]: Commented out for testing?

File: src/libsync/owncloudtheme.cpp [hunk 6]
```diff
@@ -48,4 +46,15 @@ QIcon ownCloudTheme::aboutIcon() const
...
     return Resources::themeUniversalIcon(QStringLiteral("oc-image-about"));
 }
+
+QmlButtonColor ownCloudTheme::brandedButtonColor() const
+{
+    const QColor button("#709cd2");
+    return {button, Qt::white, button.darker()};
+}
+
+QmlButtonColor ownCloudTheme::brandedButtonColor2() const
```
Comment (erikjv) [inline src/libsync/owncloudtheme.cpp:56]: 2?

**Review by erikjv — CHANGES_REQUESTED** [review]

File: src/gui/creds/qmlcredentials.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/gui/creds/qmlcredentials.cpp:86]: This is QmlBasicCredentials which depends on user name and password.

File: src/gui/creds/qmlcredentials.cpp (hunk 3 above)
Comment (TheOneRing) [reply src/gui/creds/qmlcredentials.cpp:100]: basic auth

File: src/gui/creds/qmlcredentials.cpp (hunk 4 above)
Comment (TheOneRing) [reply src/gui/creds/qmlcredentials.cpp:117]: basic auth

File: src/libsync/owncloudtheme.cpp (hunk 6 above)
Comment (TheOneRing) [reply src/libsync/owncloudtheme.cpp:56]: Its defining the color for the less important login button, so just a  second pair of colors.
No idea how to call that.

Comment (erikjv) [reply src/libsync/owncloudtheme.cpp:56]: Maybe `primaryColor` and `secondaryColor`?

File: src/libsync/theme.h [hunk 7]
```diff
@@ -57,6 +76,10 @@ class OWNCLOUDSYNC_EXPORT Theme : public QObject
     Q_PROPERTY(bool syncNewlyDiscoveredSpaces READ syncNewlyDiscoveredSpaces FINAL CONSTANT)
     Q_PROPERTY(QColor avatarColor READ avatarColor NOTIFY themeChanged)
     Q_PROPERTY(QColor avatarColorChecked READ avatarColorChecked NOTIFY themeChanged)
+    Q_PROPERTY(QColor brandedBackgoundColor READ wizardHeaderBackgroundColor CONSTANT)
```
Comment (DeepDiver1975) [inline src/libsync/theme.h:79]: what about dropping 'branded' here? anything in a theme is about branding - right?

**Review by erikjv — APPROVED** [review]

**Comment by saw-jan** [comment]
> @saw-jan this change broke the tests, please change the tests.

Thank you for letting us know. we will change the test behavior.

**Pull request merged**